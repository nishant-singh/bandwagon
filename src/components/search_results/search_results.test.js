import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';

import SearchResults  from './search_results_container';


/**
 * Function to create a Shallow Wrapper for the SearchResults component.
 */
const setup = (initState, toMount) => {
  const mockStore = configureStore(),
        store = mockStore(initState);
  if(toMount)
    return mount(<SearchResults store = {store} artist={initState.searchedArtist} />);
  else
    return shallow(<SearchResults store = {store} artist={initState.searchedArtist} />);
};

/**
 * Return Shallow Wrapper containing node(s) with the given data-test value.
 */

describe('SearchResult', () => {
  let wrapper;
  const artist = {
    facebook_page_url: "https://www.facebook.com/maroon5",
    id: "510",
    image_url: "https://s3.amazonaws.com/bit-photos/large/8479721.jpeg",
    name: "Maroon 5",
    thumb_url: "https://s3.amazonaws.com/bit-photos/thumb/8479721.jpeg",
    events : [
      {
        venue: {
          city: "Toronto",
          country: "Canada",
          latitude: "43.643466",
          longitude: "-79.379142",
          name: "Air Canada Centre"
        },
        datetime: '2018-09-07T21:00:00',
        id: '1011359000',
        artist_id: '139493'
      },
    ]
  };

  beforeEach(() => {
    wrapper = setup({
      searchedArtist: artist
    });
  });
  test('should render correctly in "debug" mode', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('renders artist cards', () => {
    let wrapper = setup({
      searchedArtist: artist
    },  true);
    const artistCard = wrapper.find('.profileCard');
    expect(artistCard.length).toBe(1);
  });

  test('renders artist\'s events', () => {
    let wrapper = setup({
      searchedArtist: artist
    }, true);
    const events = wrapper.find('.eventList');
    expect(events.length).toBe(1);
  });
});