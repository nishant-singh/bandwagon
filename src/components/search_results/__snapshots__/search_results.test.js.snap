// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(SearchResults)
    artist={
        Object {
            "events": Array [
              Object {
                "artist_id": "139493",
                "datetime": "2018-09-07T21:00:00",
                "id": "1011359000",
                "venue": Object {
                  "city": "Toronto",
                  "country": "Canada",
                  "latitude": "43.643466",
                  "longitude": "-79.379142",
                  "name": "Air Canada Centre",
                },
              },
            ],
            "facebook_page_url": "https://www.facebook.com/maroon5",
            "id": "510",
            "image_url": "https://s3.amazonaws.com/bit-photos/large/8479721.jpeg",
            "name": "Maroon 5",
            "thumb_url": "https://s3.amazonaws.com/bit-photos/thumb/8479721.jpeg",
          }
    }
    store={
        Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "artist": Object {
        "events": Array [
          Object {
            "artist_id": "139493",
            "datetime": "2018-09-07T21:00:00",
            "id": "1011359000",
            "venue": Object {
              "city": "Toronto",
              "country": "Canada",
              "latitude": "43.643466",
              "longitude": "-79.379142",
              "name": "Air Canada Centre",
            },
          },
        ],
        "facebook_page_url": "https://www.facebook.com/maroon5",
        "id": "510",
        "image_url": "https://s3.amazonaws.com/bit-photos/large/8479721.jpeg",
        "name": "Maroon 5",
        "thumb_url": "https://s3.amazonaws.com/bit-photos/thumb/8479721.jpeg",
      },
      "dispatch": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "artist": Object {
          "events": Array [
            Object {
              "artist_id": "139493",
              "datetime": "2018-09-07T21:00:00",
              "id": "1011359000",
              "venue": Object {
                "city": "Toronto",
                "country": "Canada",
                "latitude": "43.643466",
                "longitude": "-79.379142",
                "name": "Air Canada Centre",
              },
            },
          ],
          "facebook_page_url": "https://www.facebook.com/maroon5",
          "id": "510",
          "image_url": "https://s3.amazonaws.com/bit-photos/large/8479721.jpeg",
          "name": "Maroon 5",
          "thumb_url": "https://s3.amazonaws.com/bit-photos/thumb/8479721.jpeg",
        },
        "dispatch": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
